UnitTests/Text/TextFormatterTests.cs

```diff
@@ -1418,5 +1604,129 @@ public void String_Array_Is_Not_Always_Equal_ToRunes_Array (string text, int run
 				Assert.Equal (expected, text [index].ToString ());
 			}
 		}
+
+		[Fact]
+		public void GetLengthThatFits_With_Combining_Runes ()
+		{
+			var text = "Les Mise\u0328\u0301rables";
+			Assert.Equal (16, TextFormatter.GetLengthThatFits (text, 14));
+		}
+
+		[Fact]
+		public void GetMaxColsForWidth_With_Combining_Runes ()
+		{
+			var text = new List<string> () { "Les Mis", "e\u0328\u0301", "rables" };
+			Assert.Equal (1, TextFormatter.GetMaxColsForWidth (text, 1));
+		}
+
+		[Fact]
+		public void GetSumMaxCharWidth_With_Combining_Runes ()
+		{
+			var text = "Les Mise\u0328\u0301rables";
+			Assert.Equal (1, TextFormatter.GetSumMaxCharWidth (text, 1, 1));
+		}
+
+		[Fact]
+		public void GetSumMaxCharWidth_List_With_Combining_Runes ()
+		{
+			var text = new List<string> () { "Les Mis", "e\u0328\u0301", "rables" };
+			Assert.Equal (1, TextFormatter.GetSumMaxCharWidth (text, 1, 1));
+		}
+
+		[Theory]
+		[InlineData (14, 1, TextDirection.LeftRight_TopBottom)]
+		[InlineData (1, 14, TextDirection.TopBottom_LeftRight)]
+		public void CalcRect_With_Combining_Runes (int width, int height, TextDirection textDirection)
+		{
+			var text = "Les Mise\u0328\u0301rables";
+			Assert.Equal (new Rect (0, 0, width, height), TextFormatter.CalcRect (0, 0, text, textDirection));
+		}
+
+		[Theory, AutoInitShutdown]
```

Comment:
Is it possible to refactor these tests to NOT use AutoInitShutdown?

IMO, TextFormatter should be able to operate completely independently of `Application`. 

If there is some reason it cannot, I consider that a bug, and we should either fix it in this PR or open another PR to do so in the future.

Reply:
Maybe I'm wrong but the `Draw` method use a drive to add runes.

Reply:
I already added this commit which fix this issue https://github.com/gui-cs/Terminal.Gui/pull/3012/commits/4fe1392886048b470751eaaa6f327a9bc75a47de